woocommerce/woocommerce-ios
# [Woo POS] Update POS alert design to match the existing app

Closes: #13016 


## Why

As the card present payment alerts presented in POS are a new SwiftUI view `CardPresentPaymentAlert` instead of the UIKit modal controller `CardPresentPaymentsModalViewController` in the app use cases, we want to match the SwiftUI view design to the UIKit version.

## How

Updated the design in `BasicCardPresentPaymentAlert` to try matching the UIKit version in `CardPresentPaymentsModalViewController` and its xib. The major difference is the content size, which I haven't found a solution using SwiftUI.sheet to customize the sheet size. I'm leaving the custom sizing as a separate subtask https://github.com/woocommerce/woocommerce-ios/issues/13016, and I'll also check with Joe about whether it's fine to use the default sheet size before having a workaround.

Previews were added for all the alert view models that I could find for Bluetooth reader connection/payment.

## Testing information


Prerequisite: the store is eligible for POS and the feature switch is enabled in Menu > Settings > Experimental Features > POS

- Launch app
- Enter POS in Menu > Point of Sale
- Add some items to cart
- Tap to check out
- Tap `Collect payment` --> the reader connection & payment alerts should be shown matching mostly to the UIKit version (collect payment from an order) except for the size of the sheet

### SwiftUI preview

Feel free to check the SwiftUI preview on `CardPresentPaymentAlert` to tap on different alert types.

## Screenshots


SwiftUI preview:

https://github.com/woocommerce/woocommerce-ios/assets/1945542/4ccadfb9-56ce-44dd-94f0-82cfe5130a13

Collect payment success flow:

https://github.com/woocommerce/woocommerce-ios/assets/1945542/7889bea2-d2d7-457e-a8ce-425c5ed02849



---
